test/pkg/environment/azure/identity.go

```
@@ -0,0 +1,94 @@
+	cluster, err := env.managedClusterClient.Get(ctx, env.ClusterResourceGroup, env.ClusterName, nil)
+	Expect(err).ToNot(HaveOccurred())
+	Expect(cluster.Identity).ToNot(BeNil())
+	Expect(cluster.Identity.PrincipalID).ToNot(BeNil())
+	return lo.FromPtr(cluster.Identity.PrincipalID)
+}
+
+func (env *Environment) GetKarpenterWorkloadIdentity(ctx context.Context) string {
+	karpenterMSIName := "karpentermsi" // matches AZURE_KARPENTER_USER_ASSIGNED_IDENTITY_NAME
+
+	cred, err := azidentity.NewDefaultAzureCredential(nil)
+	Expect(err).ToNot(HaveOccurred())
+
+	msiClient, err := armmsi.NewUserAssignedIdentitiesClient(env.SubscriptionID, cred, nil)
+	Expect(err).ToNot(HaveOccurred())
+
+	identity, err := msiClient.Get(ctx, env.ClusterResourceGroup, karpenterMSIName, nil)
+	Expect(err).ToNot(HaveOccurred())
+
+	return lo.FromPtr(identity.Properties.PrincipalID)
+}
+
+// getCurrentUserPrincipalID gets the principal ID of the current authenticated user
+func (env *Environment) GetCurrentUserPrincipalID(ctx context.Context, cred *azidentity.DefaultAzureCredential) string {
+	token, err := cred.GetToken(ctx, policy.TokenRequestOptions{
+		Scopes: []string{"https://management.azure.com/.default"},
+	})
+	if err != nil {
+		fmt.Printf("Warning: Could not get token to determine current user: %v\n", err)
+		return ""
+	}
+
+	// Parse the JWT token to extract the oid (object ID) claim
+	parser := jwt.NewParser(jwt.WithoutClaimsValidation())
+	parsedToken, _, err := parser.ParseUnverified(token.Token, jwt.MapClaims{})
+	if err != nil {
+		fmt.Printf("Warning: Could not parse JWT token: %v\n", err)
+		return ""
+	}
+
+	claims, ok := parsedToken.Claims.(jwt.MapClaims)
+	if !ok {
+		fmt.Printf("Warning: Could not extract claims from token\n")
+		return ""
+	}
+
+	// Extract the oid (object ID) claim which is the principal ID
+	if oid, ok := claims["oid"].(string); ok {
+		return oid
+	}
+
+	fmt.Printf("Warning: Could not find oid claim in token\n")
```

Feels like this method should just return an error?

Or do `Expect` asserts?